src/app/3d/qgs3danimationwidget.cpp

```diff
@@ -175,6 +188,45 @@ void Qgs3DAnimationWidget::onAnimationTimer()
   }
 }
 
+void Qgs3DAnimationWidget::onExportAnimation()
+{
+  if ( !mMap || !mAnimationSettings )
+    QMessageBox::warning( this, tr( "Export Animation" ), tr( "Unable to export 3D animation" ) );
+
+  Qgs3DAnimationExportDialog dialog;
+  if ( dialog.exec() == QDialog::Accepted )
+  {
+    QgsFeedback progressFeedback;
+
+    QProgressDialog progressDialog( tr( "Exporting frames..." ), tr( "Abort" ), 0, 100, this );
+    progressDialog.setWindowModality( Qt::WindowModal );
+    QString error;
+
+    connect( &progressFeedback, &QgsFeedback::progressChanged, this,
+             [&progressDialog, &progressFeedback] { progressDialog.setValue( static_cast<int>( progressFeedback.progress() ) ); }
+           );
+
+    connect( &progressDialog, &QProgressDialog::canceled, this,
+             [&progressFeedback] { progressFeedback.cancel(); }
+           );
```

Comment:
cancel() is a slot so in fact we could connect QProgressDialog::canceled directly to it - without an extra lambda function